docs/docs/path-prefix.md

```diff
@@ -2,46 +2,82 @@
 
-E.g. a Gatsby blog could live at `example.com/blog/` or a site could be hosted
-on GitHub Pages at `example.github.io/my-gatsby-site/`
+For example, a Gatsby blog could live at `example.com/blog/` or a site could be hosted on GitHub Pages at `example.github.io/my-gatsby-site/`
 
 Each of these sites need a prefix added to all paths on the site. So a link to
 `/my-sweet-blog-post/` should be rewritten to `/blog/my-sweet-blog-post`.
 
-In addition links to various resources (JavaScript, images, CSS) need the same
-prefix added (this is accomplished by setting the `publicPath` in webpack).
+In addition links to various resources (JavaScript, CSS, images, and other static content) need the same prefix, so that the site continues to function and display correctly, even if served from this path prefix.
 
-Luckily, for most sites, this work can be offloaded to Gatsby. Using
-[Gatsby's Link component](/docs/gatsby-link/) for internal links ensures those links
-will be prefixed correctly. Gatsby ensures that paths created internally and by
-webpack are also correctly prefixed.
+Let's get this functionality implemented. We'll add an option to our `gatsby-config.js`, and add a flag to our build command.
 
-## Development
+### Add to `gatsby-config.js`
 
-During development, write paths as if there was no path prefix e.g. for a blog
-hosted at `example.com/blog`, don't add `/blog` to your links. The prefix will
-be added when you build for deployment.
-
-## Production build
-
-There are two steps for building a site with path prefixes.
-
-First define the prefix in your site's `gatsby-config.js`.
-
-```javascript:title=gatsby-config.js
+```js:title=gatsby-config.js
 module.exports = {
-  // Note: it must *not* have a trailing slash.
   pathPrefix: `/blog`,
 }
 ```
 
-Then pass `--prefix-paths` cmd option to Gatsby.
+### Build
+
+Once the `pathPrefix` is specified in `gastby-config.js`, we are well on our way to a prefixed app. The final step is to build out your application with a flag `--prefix-paths`, like so:
 
 ```shell
 gatsby build --prefix-paths
 ```
 
-NOTE: When running the command without the `--prefix-paths` flag, Gatsby ignores
-your `pathPrefix`.
+If this flag is not passed, Gatsby will ignore your `pathPrefix` and build out your site as if it were hosted from the root domain.
+
+### In-app linking
+
+As a developer using this feature, it should be seamless. We provide APIs and libraries to make using this functionality a breeze. Specifically, the [`Link`](/docs/gatsby-link/) component has built-in functionality to handle path prefixing.
+
+For example, if we want to link to our `/page-2` link (but the actual link will be prefixed, e.g. `/blog/page-2`) we don't want to hard code this path prefix in all of our links. We have your back! Merely by using the `Link` component, we will automatically prefix your paths for you. If you later migrate off of `pathPrefix` your links will _still_ work seamlessly.
```

Comment:
No huge deal, but should we avoid words like `merely`? (here and later in same docs). How about
```suggestion
For example, if we want to link to our `/page-2` link (but the actual link will be prefixed, e.g. `/blog/page-2`) we don't want to hard code this path prefix in all of our links. We have your back! By using the `Link` component, we will automatically prefix your paths for you. If you later migrate off of `pathPrefix` your links will _still_ work seamlessly.
```